### PR TITLE
login, lobby, rooms, and games

### PR DESCRIPTION
**Pull request by SylvanSign**
she's a big'n probably best if we try it out during our meetup

Fixes issues:
- https://github.com/ZakMiller/TheEmpire/issues/1
- https://github.com/ZakMiller/TheEmpire/issues/11
- https://github.com/ZakMiller/TheEmpire/issues/12

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -7,5 +7,8 @@ module.exports = {
   "plugins": [
     "standard",
     "promise"
-  ]
+  ],
+  "rules": {
+    "space-before-function-paren": ["error", "never"]
+  }
```
Comment (SylvanSign) [inline .eslintrc.js:16]: this is the style change I like @ZakMiller

File: .gitignore [hunk 2]
```diff
@@ -1,3 +1,4 @@
 node_modules/
+.vscode/
```
Comment (SylvanSign) [inline .gitignore:2]: Visual Studio Code debug launcher configs shouldn't be source controlled

**Review by SylvanSign — COMMENTED** [review]
I'll go over this with y'all Thursday evening

File: public/scripts/main.js [hunk 3]
```diff
@@ -1,43 +1,149 @@
...
+  const countdownBanner = document.querySelector('#countdownBanner')
+
+  let timeoutReference
+
+  // Event Listeners
+  function updateLobby(usernames) {
+    lobbyList.innerHTML = ''
+    usernames.forEach(name => {
+      const listItem = document.createElement('li')
+      const text = (username === name) ? `> ${name}` : name
```
Comment (ZakMiller) [inline public/scripts/main.js]: What's going on here? Does someone have a username AND a name?

Comment (SylvanSign) [reply public/scripts/main.js]: I told you variable names are hard... `username` refers to this particular user's username, `usernames` refers to the list of all users that the server sent us. And `name` refers to each name in that usernames list as we iterate over it. If the current name in the list I'm looking at is my name, add a special style to it basically. Maybe I can make the naming clearer

Comment (SylvanSign) [reply public/scripts/main.js]: Maybe I can store the client's name as `myUsername` or something

Comment (SylvanSign) [reply public/scripts/main.js]: Also, I should have this function after the registration one to make it clearer

File: rooms.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+      name = ''
+      for (let i = 0; i !== length; ++i) {
+        name += generateRandomChar()
+      }
+    } while (roomNames.has(name))
+
+    roomNames.add(name)
+    return name
+  }
+}
```
Comment (SylvanSign) [inline rooms.js:26]: This module was created with custom games in mind. So you could generate random room names that would be pretty easy to manually type in, like Quiplash does it.

Comment (SylvanSign) [reply rooms.js:26]: Eventually we need to add logic to remove dead game rooms from the set so we don't eventually run out of unique room names.

File: public/scripts/main.js (hunk 3 above)
Comment (ZakMiller) [reply public/scripts/main.js]: How about usernames.forEach(username

and ya, make username myUsername

Comment (SylvanSign) [reply public/scripts/main.js]: pushed a commit to address this

**Review by ZakMiller — APPROVED** [review]

**Pull request merged**